simphony/io/tests/test_file_mesh.py

```diff
@@ -0,0 +1,603 @@
+        """ Check that a cell can be added correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cells = [
+            Cell(puids[0:4])
+        ]
+
+        self.mesh.add_cell(cells[0])
+        self.assertRaises(KeyError, self.mesh.add_cell, cells[0])
+
+    def test_non_emtpy_edges(self):
+        """ Checks that the list of edges is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:2]]
+
+        edge = Edge(puids[0:2])
+
+        self.mesh.add_edge(edge)
+
+        self.assertTrue(self.mesh.has_edges())
+
+    def test_non_emtpy_faces(self):
+        """ Checks that the list of faces is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        face = Face(puids[0:3])
+
+        self.mesh.add_face(face)
+
+        self.assertTrue(self.mesh.has_faces())
+
+    def test_non_emtpy_cells(self):
+        """ Checks that the list of cells is not empty
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cell = Cell(puids[0:4])
+
+        self.mesh.add_cell(cell)
+
+        self.assertTrue(self.mesh.has_cells())
+
+    def test_get_point(self):
+        """ Check that a point can be retrieved correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:1]]
+
+        point_ret = self.mesh.get_point(puids[0])
+
+        self.assertTrue(isinstance(point_ret, Point))
+        self.assertEqual(puids[0], point_ret.uid)
+
+    def test_get_edge(self):
+        """ Check that an edge can be retrieved correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:2]]
+
+        edges = [
+            Edge(puids[:]),
+        ]
+
+        euids = [self.mesh.add_edge(edge) for edge in edges]
+
+        edge_ret = self.mesh.get_edge(euids[0])
+
+        self.assertTrue(isinstance(edge_ret, Edge))
+        self.assertEqual(euids[0], edge_ret.uid)
+
+    def test_get_face(self):
+        """ Check that a face can be retrieved correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        faces = [
+            Face(puids[:])
+        ]
+
+        fuids = [self.mesh.add_face(face) for face in faces]
+
+        face_ret = self.mesh.get_face(fuids[0])
+
+        self.assertTrue(isinstance(face_ret, Face))
+        self.assertEqual(fuids[0], face_ret.uid)
+
+    def test_get_cell(self):
+        """ Check that a cell can be retrieved correctly
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:4]]
+
+        cells = [
+            Cell(puids[:])
+            ]
+
+        cuids = [self.mesh.add_cell(cell) for cell in cells]
+
+        cell_ret = self.mesh.get_cell(cuids[0])
+
+        self.assertTrue(isinstance(cell_ret, Cell))
+        self.assertEqual(cuids[0], cell_ret.uid)
```

Comment:
similar comment to https://github.com/simphony/simphony-common/pull/36/files#r24573824